rust/extns/numaflow-sqs/src/sink.rs

```diff
@@ -0,0 +1,270 @@
+        // Validate required fields
+        if self.region.is_empty() {
+            return Err(Error::InvalidConfig("region is required".to_string()));
+        }
+
+        if self.queue_name.is_empty() {
+            return Err(Error::InvalidConfig("queue name is required".to_string()));
+        }
+
+        if self.queue_owner_aws_account_id.is_empty() {
+            return Err(Error::InvalidConfig(
+                "queue owner AWS account ID is required".to_string(),
+            ));
+        }
+
+        Ok(())
+    }
+}
+
+/// Creates and configures an SQS client for sink based on the provided configuration.
+pub async fn create_sqs_client(config: Option<SqsSinkConfig>) -> Result<Client> {
+    tracing::info!(
+        "Creating SQS sink client for queue {queue_name} in region {region}",
+        region = config
+            .as_ref()
+            .map(|c| c.region.clone())
+            .unwrap_or_default(),
+        queue_name = config
+            .as_ref()
+            .map(|c| c.queue_name.clone())
+            .unwrap_or_default()
+    );
+
+    crate::create_sqs_client(config.map(crate::SqsConfig::Sink)).await
+}
+
+enum SqsSinkActorMessage {
+    SendMessageBatch {
+        respond_to: oneshot::Sender<Result<Vec<SqsSinkResponse>>>,
+        messages: Vec<SqsSinkMessage>,
+    },
+}
+
+pub struct SqsSinkMessage {
+    pub id: String,
+    pub message_body: Bytes,
+}
+
+struct SqsSinkActor {
+    handler_rx: mpsc::Receiver<SqsSinkActorMessage>,
+    client: Client,
+    queue_url: String,
+}
+
+impl SqsSinkActor {
+    pub fn new(
+        handler_rx: mpsc::Receiver<SqsSinkActorMessage>,
+        client: Client,
+        queue_url: String,
+    ) -> Self {
+        Self {
+            handler_rx,
+            client,
+            queue_url,
+        }
+    }
+
+    async fn run(&mut self) {
+        while let Some(msg) = self.handler_rx.recv().await {
+            self.handle_message(msg).await;
+        }
+    }
+
+    async fn handle_message(&self, msg: SqsSinkActorMessage) {
+        match msg {
+            SqsSinkActorMessage::SendMessageBatch {
+                respond_to,
+                messages,
+            } => {
+                let sink_responses = self.sink_message_batch(messages).await;
+                respond_to
+                    .send(sink_responses)
+                    .expect("failed to send response from SqsSinkActorMessage::SendMessageBatch");
+            }
+        }
+    }
+
+    async fn sink_message_batch(
+        &self,
+        messages: Vec<SqsSinkMessage>,
+    ) -> Result<Vec<SqsSinkResponse>> {
+        let mut entries = Vec::with_capacity(messages.len());
+        let mut sink_message_responses = Vec::with_capacity(messages.len());
+        for message in messages {
+            let entry = SendMessageBatchRequestEntry::builder()
+                .id(message.id)
+                .message_body(String::from_utf8_lossy(&message.message_body).to_string())
```

Comment:
sqs allows you to create only String type messages?

Reply:
Yes only strings are allowed [docs](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_SendMessageBatchRequestEntry.html)